mkosi/util.py

```diff
@@ -269,3 +269,7 @@ def mandatory_variable(name: str) -> str:
         return os.environ[name]
     except KeyError:
         die(f"${name} must be set in the environment")
+
+
+def dir_only_contains(dir: Path, tgt: PathString) -> bool:
+    return not dir.is_dir() or (dir / tgt).exists() and len(list(dir.glob("*"))) == 1
```

Comment:
Instead of the list allocation and eager glob, the following should also do the trick and be more efficient:

```python
try:
    [child] = dir.iterdir()
except ValueError:
    return False # unwrapping failed, either 0 or 2+ children

Now that it is extracted into a function there is no need to keep it a oneliner

Reply:
This is indeed prettier. The whole function becomes a bit unwieldy, but at least it's twice as fast!